packages/offix-datastore/src/models.ts

```diff
@@ -0,0 +1,21 @@
+import { GraphQLSchema, GraphQLObjectType } from "graphql";
+import { parseMarker } from "graphql-metadata";
+import { getUserTypesFromSchema } from "@graphql-toolkit/common";
+
+export interface Model {
+    __typename: string;
+}
+
+export interface PersistedModel extends Model {
+    id: string;
+}
+
+export function extractModelsFromSchema(schema: GraphQLSchema): Model[] {
+    const types = getUserTypesFromSchema(schema);
+    const userTypes = types.filter((modelType: GraphQLObjectType) => parseMarker("model", modelType.description));
```

Comment:
Nice! I like how you can enable model processing. Cannot comprehend how many good stuff this have. I think this comes back as very strongly tied to graphback. To not tie it I think we can:

- In the graphback client we can generate models file for offix datastore. 
- For offix we going to document how to create models files by themselves so you do not need schema and very performance intensive operations happening on the client - it will be simple JSON

WDYT?

Reply:
Yes! This is great. I was thinking codegen before but using graphback client is better. Another reason I check schema though is to get the object stores to create and delete. The user could provide a list of models instead.